pkg/ccl/sqlproxyccl/forwarder.go

```diff
@@ -495,6 +507,9 @@ func (p *processor) resume(ctx context.Context) error {
 	}
 
 	if err := enterResume(); err != nil {
+		if errors.Is(err, errProcessorResumed) {
```

Comment:
I don't understand the intent behind this change. Why do we return nil here? Is there something about the PR that introduced a new case to check, or is this cleaning up misbehavior that already existed?

Reply:
This is to clean up an issue that already existed. `resumeProcessors` may be called concurrently when the connection started + when a transfer gets triggered. When that happens, resume returns errProcessorResumed, which causes the forwarder to be closed. 

> // resumeProcessors starts both the request and response processors
// asynchronously. The forwarder will be closed if any of the processors
// return an error while resuming. This is idempotent as resume() will return
// nil if the processor has already been started.

The comment states that resume should return nil if the processor has already been started.